gammapy/datasets/map.py

```diff
@@ -2506,16 +2508,41 @@ def update(self, exposure, psf, edisp, geom):
             Edisp map.
         geom : `WcsGeom`
             Counts geom
+        mask_safe_psf : `~gammapy.maps.Map`
+            Mask safe map of boolean type.
         """
         # TODO: simplify and clean up
         log.debug("Updating model evaluator")
         # cache current position of the model component
 
+        if mask_safe_psf is None:
+            mask = exposure.copy()
+            mask.data = (exposure.data != 0) & np.isfinite(exposure.data)
+        else:
+            mask = mask_safe_psf
+        mask = mask.reduce_over_axes(func=np.logical_or)
+
+        if (
+            self.model.position is not None
+            and isinstance(mask.geom, WcsGeom)
+            and not mask.geom.contains(self.model.position)[0]
```

Comment:
I don't think this condition is correct, because the model can be at a position inside the geometry, but where no IRFs are defined because of the FoV cut. I think in this case we still want to use the nearest valid IRF...